### PR TITLE
Add org.tuxfamily.StuntRally

### PR DESCRIPTION
**Pull request by AsciiWolf**

### CODE REVIEW

**Comment by AsciiWolf** [comment]
This is a work in progress. Some dependencies (for example OGRE) and game data were not yet added. They will be added later, after I fix the current issue - compilation fails on "cannot find -lpthreads"; it looks like that the pthread lib/headers is not available in the sdk runtime or not found by CMake.

**Comment by AsciiWolf** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by AsciiWolf** [comment]
That's weird. It looks like that there was no issue with pthreads on the Flathub [build server](https://flathub.org/builds/#/builders/30/builds/149), so it's probably some issue with the flatpak-builder on my Fedora machine.

**Comment by AsciiWolf** [comment]
Is there any way to download the _CMakeOutput.log_ and _CMakeError.log_ files from the build server?

**Comment by TingPing** [comment]
Sadly not.

**Comment by AsciiWolf** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by nedrichards** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by nedrichards** [comment]
It'd be great to send that appdata upstream, if possible.

**Comment by AsciiWolf** [comment]
@nedrichards I will. But upstream seems pretty dead and the last stable release was in 2015.

**Comment by nedrichards** [comment]
Sad times. But yeah, still worthwhile.

**Comment by AsciiWolf** [comment]
By the way, this is still WIP. Currently crashing on some code errors during compilation. I will try to fix them when I have some time.

File: org.tuxfamily.StuntRally.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            ]
+        },
+        {
+            "name": "MyGUI",
+            "buildsystem": "cmake",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/MyGUI/mygui.git",
+                    "commit": "3687a78139837b598fbe703f2833add4c39adcca"
```
Comment (AsciiWolf) [inline org.tuxfamily.StuntRally.json]: I had to use a commit from master here. The last stable version of MyGUI is from 2015, is not compatible with newer Ogre3D and fails to compile. The stable version would have to be heavily patched to make it working correctly. :-/

**Comment by AsciiWolf** [comment]
It looks like that I am not able to fix the [compilation issue](https://flathub.org/builds/#/builders/32/builds/253/steps/5/logs/stdio). The game is probably not compatible with newer Ogre3D engine builds, but older version of Ogre3D fails to compile as well. :-( I am, unfortunately, not so skilled in C. Anyone has any idea how to fix it?

**Comment by nedrichards** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by nedrichards** [comment]
Build fails in sdl4ogre https://flathub.org/builds/#/builders/2/builds/2476

**Comment by AsciiWolf** [comment]
I have tried to fix the build with [my friend](https://github.com/Zlopez) who wrote a patch.

Unfortunately, it now fails to find a Bullet library:
```
CMake Error at cmake/FindPackageHandleStandardArgs.cmake:91 (MESSAGE):
  Could NOT find Bullet (missing: BULLET_FILELOADER_LIBRARY
  BULLET_WORLDIMPORTER_LIBRARY)
```

**Comment by nedrichards** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by AsciiWolf** [comment]
I have pushed another update. I have fixed the CMake Error and the compilation now starts, unfortunately it crashes on some Bullet-related undefined reference errors. Also, the game data were not yet added, they will be added after I fix the current issue.

**Comment by nedrichards** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by AsciiWolf** [comment]
I have fixed the errors. The Flatpak now compiles and runs, but crashes when trying to load any racing track.

**Comment by AsciiWolf** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by AsciiWolf** [comment]
stuntrally

I finally got it working! There were some issues with the OIS module and Boost library that I previously didn't notice.

**Comment by AsciiWolf** [comment]
bot, build org.tuxfamily.StuntRally

**Comment by barthalion** [comment]
Cool! Is it good to merge now? Build passed in buildbot.

**Comment by AsciiWolf** [comment]
@barthalion Yup, it seems to work fine. :-)

File: org.tuxfamily.StuntRally.json [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://dl.bintray.com/boostorg/release/1.63.0/source/boost_1_63_0.tar.gz",
+                    "sha256": "fe34a4e119798e10b8cc9e565b3b0284e9fd3977ec8a1b19586ad1dec397088b"
+                }
+            ],
+            "build-commands": [
+                "./bootstrap.sh --prefix=/app --with-libraries=system,filesystem,thread,wave",
+                "./b2 install variant=release link=shared runtime-link=shared -j `nproc`"
```
Comment (barthalion) [inline org.tuxfamily.StuntRally.json]: `$FLATPAK_BUILDER_N_JOBS` instead of nproc

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.tuxfamily.StuntRally

Thanks!

**Pull request closed without merging**